### PR TITLE
w3c phase 2: add last parent_id to tracestate

### PR DESCRIPTION
**Pull request by mabdinur**
### What does this PR do?

With this change span_id will be encoded in both the tracestate and traceparent headers. This will allow us to re-connect datadog traces that contain non-datadog spans. 

Ex: Host A: Datadog Spans ->tracecontext-> Host B: Otel Dynatrace Spans -->tracecontext-> Host C: Otel Datadog Spans
By adding span_id to the tracestate Host C will be able to detect upstream Datadog spans from Host A. This span_id will then be sent to Datadog intake services (via `_dd.parent_id` tag). The Datadog internal services will use this tag to reconnect Datadog Spans.

Tests will be added in this PR: https://github.com/DataDog/system-tests/pull/2181

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### Security 
Datadog employees:
- [ ] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

**Review by khanayan123 — COMMENTED** [review]
LGTM, but I will have to defer the approval of this PR to @rochdev since he might have some opinions on changes to tracestate

**Review by rochdev — APPROVED** [review]

**Comment by mabdinur** [comment]
Blocked by: https://github.com/DataDog/dd-trace-js/pull/4153

File: packages/dd-trace/src/opentracing/propagation/text_map.js
Comment (mabdinur) [inline packages/dd-trace/src/opentracing/propagation/text_map.js:174]:
```suggestion
     if (!spanContext._isRemote){
        # SpanContext was created by a ddtrace span. 
        # Last datadog span id should be set to the current span.
        state.set('p', spanContext._spanId)
     } else if (spanContext._trace.tags['_dd.parent_id']) {
        # Propagate the last Datadog span id set on the remote span.
        state.set('p', spanContext._trace.tags['_dd.parent_id'])
     }
```

Comment (Qard) [reply packages/dd-trace/src/opentracing/propagation/text_map.js:174]: Wrong comment syntax. 😅 

```suggestion
      if (!spanContext._isRemote) {
        // SpanContext was created by a ddtrace span. 
        // Last datadog span id should be set to the current span.
        state.set('p', spanContext._spanId)
      } else if (spanContext._trace.tags['_dd.parent_id']) {
        // Propagate the last Datadog span id set on the remote span.
        state.set('p', spanContext._trace.tags['_dd.parent_id'])
      }
```

**Review by rochdev — CHANGES_REQUESTED** [review]
Can a test for this be added?

**Review by khanayan123 — APPROVED** [review]

**Review by rochdev — APPROVED** [review]

**Pull request merged**
